common/Helpers/GPOHelper.cs

```diff
@@ -0,0 +1,82 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Microsoft.Win32;
+
+namespace DevHome.Common.Helpers;
+
+public class GPOHelper
+{
+    private enum GpoRuleConfigured
+    {
+        WrongValue = -3, // The policy is set to an unrecognized value
+        Unavailable = -2, // Couldn't access registry
+        NotConfigured = -1, // Policy is not configured
+        Disabled = 0, // Policy is disabled
+        Enabled = 1, // Policy is enabled
+    }
+
+    // Registry path where gpo policy values are stored
+    private const string PoliciesScopeMachine = "HKEY_LOCAL_MACHINE";
+    private const string PoliciesPath = @"\SOFTWARE\Policies\DevHome";
+
+    // Registry value names
+    private const string PolicyConfigureEnabledDevHome = "ConfigureEnabledDevHome";
+
+    private static GpoRuleConfigured GetConfiguredValue(string registryValueName)
+    {
+        try
+        {
+            var rawValue = Registry.GetValue(
+                keyName: PoliciesScopeMachine + PoliciesPath,
+                valueName: registryValueName,
+                defaultValue: GpoRuleConfigured.NotConfigured);
+
+            // Value will be null if the subkey specified by keyName does not exist.
+            if (rawValue == null)
+            {
+                return GpoRuleConfigured.NotConfigured;
+            }
+            else if (rawValue is not int)
+            {
+                return GpoRuleConfigured.WrongValue;
+            }
+            else
+            {
+                return (GpoRuleConfigured)rawValue;
+            }
+        }
+        catch (System.Security.SecurityException)
+        {
+            // The user does not have the permissions required to read from the registry key.
+            return GpoRuleConfigured.Unavailable;
+        }
+        catch (System.IO.IOException)
+        {
+            // The RegistryKey that contains the specified value has been marked for deletion.
+            return GpoRuleConfigured.Unavailable;
+        }
+        catch (System.ArgumentException)
+        {
+            // keyName does not begin with a valid registry root.
+            return GpoRuleConfigured.NotConfigured;
+        }
+    }
+
+    private static bool EvaluateConfiguredValue(string registryValueName, GpoRuleConfigured defaultValue)
+    {
+        var configuredValue = GetConfiguredValue(registryValueName);
+        if (configuredValue < 0)
+        {

```

Comment:
Should add a log entry here logging the evaluated value and that the default is being used since these are extraordinary cases. If it is supposed to be disabled but isn't because the value couldn't be read or was malformed, that would be helpful for troubleshooting to have the reason in the log.